packages/prompts/src/progress-bar.ts

```diff
@@ -0,0 +1,70 @@
+import type { State } from '@clack/core';
+import color from 'picocolors';
+import { unicodeOr } from './common.js';
+import { type SpinnerOptions, type SpinnerResult, spinner } from './spinner.js';
+
+const S_PROGRESS_CHAR: Record<NonNullable<ProgressOptions['style']>, string> = {
+	light: unicodeOr('─', '-'),
+	heavy: unicodeOr('━', '='),
+	block: unicodeOr('█', '#'),
+};
+
+export interface ProgressOptions extends SpinnerOptions {
+	style?: 'light' | 'heavy' | 'block';
+	max?: number;
+	size?: number;
+}
+
+export interface ProgressResult extends SpinnerResult {
+	advance(step?: number, msg?: string): void;
+}
+
+export function progress({
+	style = 'heavy',
+	max = 100,
+	size = 40,
+	...spinnerOptions
+}: ProgressOptions = {}): ProgressResult {
+	const spin = spinner(spinnerOptions);
+	let value = 0;
+	let previousMessage = '';
+
+	const _max = Math.max(1, max);
```

Comment:
since a name prefixed with `_` usually means "ignore this being unused", maybe we should name these two `max` and `size`. then name the inputs, `userMax` and `userSize` or something.

```ts
export function progress({
  max: userMax = 100,
  size: userSize = 40
```